app/Http/Controllers/PostController.php

```diff
@@ -2,64 +2,118 @@
 
 namespace App\Http\Controllers;
 
+use App\Contracts\Services\PostServiceInterface;
+use App\Contracts\Services\UserServiceInterface;
+use App\Http\Requests\StorePostRequest;
+use App\Http\Requests\UpdatePostRequest;
 use App\Models\Post;
-use Illuminate\Http\Request;
+use Illuminate\View\View;
+use Illuminate\Support\Facades\Auth;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
 
 class PostController extends Controller
 {
+    protected $postService;
+    protected $userService;
+    public function __construct(PostServiceInterface $postService, UserServiceInterface $userService)
+    {
+        $this->postService = $postService;
+        $this->userService = $userService;
+    }
+
     /**
      * Display a listing of the resource.
+     *
+     * @return View
      */
-    public function index()
+    public function index(): View
     {
-        //
+        $posts = $this->postService->getAllPost();
+        return view('posts.index', ['posts' => $posts]);
     }
 
     /**
      * Show the form for creating a new resource.
+     *
+     * @return mixed
      */
-    public function create()
+    public function create(): mixed
     {
-        //
+        if (Auth::check() && Auth::user()) {
+            return view('posts.create', ['user_id' => Auth::user()->id]);
+        } else {
+            return redirect()->route('posts.index')->with('error', 'Must be logged into create posts.');
+        }
     }
 
     /**
      * Store a newly created resource in storage.
+     *
+     * @param StorePostRequest $request
+     * @return RedirectResponse
      */
-    public function store(Request $request)
+    public function store(StorePostRequest $request): RedirectResponse
     {
-        //
+        $post_id = $this->postService->insert($request);
+
+        return redirect()->route('posts.show', [$post_id])->with('success', 'Post created successfully');
     }
 
     /**
      * Display the specified resource.
+     *
+     * @param Post $post
+     * @return View
      */
-    public function show(Post $post)
+    public function show(Post $post): View
```

Comment:
Isn't the parameter type is `int`? 
Please check also for edit and delete function.

Reply:
https://laravel.com/docs/10.x/routing#route-model-binding

implemented as Model instead of Int so that it shows an 404 instead of an error.